samples/bluetooth/central_hr/src/main.c

```diff
@@ -221,6 +345,121 @@ static struct bt_conn_cb conn_callbacks = {
 	.disconnected = disconnected,
 };
 
+#ifdef TRACK_VND
+void passkey_entry(struct bt_conn *conn)
+{
+	printk("input key: 111111\n");
+	bt_conn_auth_passkey_entry(conn, 111111);
+}
```

Comment:
Wouldn't the correct thing to do be to use the `display_passkey` callback instead? passkey_entry should be only used for real human input.

Reply:
The intention is to allow the central_hr to do its original task (which was to continuously scan for peripherals that advertise HRS service and subscribe to their notification) also on a characteristic that requires connection security. The sample "peripheral" includes such vendor characteristic and the central_hr scans for it. To make the two samples pair, it is needed to input into the central the passkey that is selected by the peripheral. Thus, the need for the "user input". The peripheral was updated to use 111111 passkey if it is compiled with FIXED_PASSKEY option.